api/src/data_migration/transformation/transform_util.py

```diff
@@ -0,0 +1,283 @@
+import logging
+from datetime import datetime
+
+from src.constants.lookup_constants import OpportunityCategory
+from src.db.models.base import TimestampMixin
+from src.db.models.opportunity_models import (
+    Opportunity,
+    OpportunityAssistanceListing,
+    OpportunitySummary,
+)
+from src.db.models.staging.forecast import TforecastHist
+from src.db.models.staging.opportunity import Topportunity, TopportunityCfda
+from src.db.models.staging.staging_base import StagingBase
+from src.db.models.staging.synopsis import Tsynopsis, TsynopsisHist
+from src.util import datetime_util
+
+from . import SourceSummary
+
+logger = logging.getLogger(__name__)
+
+OPPORTUNITY_CATEGORY_MAP = {
+    "D": OpportunityCategory.DISCRETIONARY,
+    "M": OpportunityCategory.MANDATORY,
+    "C": OpportunityCategory.CONTINUATION,
+    "E": OpportunityCategory.EARMARK,
+    "O": OpportunityCategory.OTHER,
+}
+
+
+def transform_opportunity(
```

Comment:
Naming nit - these functions (down to `transform_opportunity_summary`) feel like main code, not "utils". Would it make sense to move them to a new file `transform_object.py` or something similar?